src/core/lib/channel/channelz_registry.cc

```diff
@@ -211,14 +227,20 @@ char* ChannelzRegistry::InternalGetServers(intptr_t start_server_id) {
 }
 
 void ChannelzRegistry::InternalLogAllEntities() {
-  MutexLock lock(&mu_);
-  for (size_t i = 0; i < entities_.size(); ++i) {
-    if (entities_[i] != nullptr) {
-      char* json = entities_[i]->RenderJsonString();
-      gpr_log(GPR_INFO, "%s", json);
-      gpr_free(json);
+  InlinedVector<RefCountedPtr<BaseNode>, 10> nodes;
+  {
+    MutexLock lock(&mu_);
```

Comment:
it's probably worth calling `reserve(entities_.size())`

Reply:
That may actually be too large if there are a lot of empty slots.  And I'm not really too worried about performance here, since this method is used solely for debugging.

Reply:
oh yes, excellent point. it's best not to :-)